Apps/GatewayApi/test/unit/Services.Test/UserFeedbackServiceTests.cs

```diff
@@ -38,92 +41,120 @@ namespace HealthGateway.GatewayApiTests.Services.Test
 
-            RequestResult<Rating> actualResult = await service.CreateRatingAsync(expectedRating);
+            // Act
+            RequestResult<RatingModel> actual = await mock.Service.CreateRatingAsync(mock.Rating);
 
-            Assert.Equal(ResultType.Success, actualResult.ResultStatus);
-            expectedRating.ShouldDeepEqual(actualResult.ResourcePayload);
+            // Assert
+            Assert.Equal(mock.Expected.ResultStatus, actual.ResultStatus);
         }
 
         /// <summary>
-        /// CreateRating - Database Error.
+        /// CreateUserFeedBack - Happy Path.
         /// </summary>
         /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
         [Fact]
-        public async Task ShouldCreateRatingWithError()
+        public async Task ShouldCreateUserFeedback()
+        {
+            // Arrange
+            CreateUserFeedbackMock mock = SetupCreateUserFeedbackMock();
+
+            // Act
+            DbResult<UserFeedback> actual = await mock.Service.CreateUserFeedbackAsync(mock.Feedback, mock.Hdid);
+
+            // Assert
+            mock.Expected.ShouldDeepEqual(actual);
+        }
+
+        private static Mock<ICryptoDelegate> GetCryptoDelegateMock()
+        {
+            Mock<ICryptoDelegate> cryptoDelegateMock = new();
+            cryptoDelegateMock.Setup(s => s.GenerateKey()).Returns(() => "Y1FmVGpXblpxNHQ3dyF6JUMqRi1KYU5kUmdVa1hwMnM=");
+            cryptoDelegateMock.Setup(s => s.Encrypt(It.IsAny<string>(), It.IsAny<string>())).Returns((string key, string text) => text + key);
+            cryptoDelegateMock.Setup(s => s.Decrypt(It.IsAny<string>(), It.IsAny<string>())).Returns((string key, string text) => text.Remove(text.Length - key.Length));
+            return cryptoDelegateMock;
+        }
+
+        private static CreateRatingMock SetupCreateRatingMock(bool dbErrorExists = false)
         {
-            Rating expectedRating = new()
+            Guid ratingId = Guid.NewGuid();
+
+            SubmitRating createRating = new()
             {
                 RatingValue = 5,
                 Skip = false,
             };
 
-            DbResult<Rating> insertResult = new()
+            Rating rating = new()
             {
-                Payload = expectedRating,
-                Status = DbStatusCode.Error,
+                Id = ratingId,
+                RatingValue = createRating.RatingValue,
+                Skip = createRating.Skip,
+            };
+
+            DbResult<Rating> dbResult = new()
+            {
+                Payload = rating,
+                Status = dbErrorExists ? DbStatusCode.Error : DbStatusCode.Created,
+            };
+
+            RequestResult<RatingModel> expected = new()
+            {
+                ResultStatus = dbErrorExists ? ResultType.Error : ResultType.Success,
+                ResourcePayload = new() { Id = ratingId, RatingValue = rating.RatingValue, Skip = rating.Skip },
             };
```

Comment:
If `dbError` is true, I'd expect the DbResult's `Message` to be set and its `Payload` to not be set.

I think both `dbResult` and `expected` could be expanded to reflect that - and the tests could then use ShouldDeepEqual().

Reply:
Changed.